doc/source/whatsnew/v0.25.0.rst

```diff
@@ -26,6 +26,44 @@ Other Enhancements
 Backwards incompatible API changes
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+GroupBy.apply on ``DataFrame`` evaluates first group only once
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The implementation of ``DataFrame.groupby.apply`` previously evaluated func
```

Comment:
see if you can make this a proper reference, I think:

``:meth:`DataFrameGroupBy.apply() <pandas.core.groupby.DataFrameGroupBy.apply>` ``

should work